### ISSUE TITLE
Pasting empty nested list and converting it to different list type throws error 

### ISSUE THREAD

**Issue by jswiderski** [ckeditor/ckeditor-dev#2411]
## Type of report

Bug

## Provide detailed reproduction steps (if any)

1. Go to https://sdk.ckeditor.com/samples/fullpreset.html.
2. Clear Editor contents and switch to source mode.
3. Paste below HTML
```
<ul>	
	<li>
	<div>test
	<ul>
	</ul>
	</div>
	</li>
	<li>test</li>
</ul>
```
In Chrome it is enough to use 

```
<ul>	
	<li>
	<div>test
	<ul>
	</ul>
	</div>
	</li>
</ul>
```
4. Switch to wysiwyg mode
5. Press `Ctrl+A` na dclick on Numbered  List button

### Expected result

No error gets thrown.

Perhaps as an alternative solution such list should be filtered by ACF.

### Actual result

JavaScript error gets thrown: 
**Message:** Cannot read property 'parent' of undefined
**Line:** 319
**URI:** `/ckeditor-dev/plugins/list/plugin.js
`
## Other details

* Browser: Any
* OS: Any
* CKEditor version: 4.0+
* Installed CKEditor plugins: List

----
The Original scenario was:
1. Open https://codepen.io/pratikgolchha/pen/wxGbRd in Edge
2. Copy everything form https://docs.microsoft.com/zh-cn/powerapps/maker/canvas-apps/controls/control-gallery and paste in the CKEditor
3. Select all contents and apply bulleted list and then numbered list
4. Check the console

This document is huge so maybe there is something extra what is causing the same issue (there shouldn't be but it may).

**Comment by jswiderski** [comment]
This issue could be related: https://github.com/ckeditor/ckeditor-dev/issues/2438

**Comment by msamsel** [comment]
Additional markup to check:
```
<p><br></p><ul></ul><ul><li><a href="https://powerusers.microsoft.com/plugins/common/feature/oauth2sso/sso_login_redirect?referer=https%3A%2F%2Fpowerusers.microsoft.com%2Ft5%2FPowerApps-Community%2Fct-p%2FPowerApps1">Register</a></li><li><br></li><li>·</li><li><br></li><li><a href="https://powerusers.microsoft.com/plugins/common/feature/oauth2sso/sso_login_redirect?referer=https%3A%2F%2Fpowerusers.microsoft.com%2Ft5%2FPowerApps-Community%2Fct-p%2FPowerApps1">Sign In</a></li><li><br></li><li>·</li><li><br></li><li><a href="https://powerusers.microsoft.com/t5/help/faqpage">Help</a></li></ul><p><br></p>
```

**Comment by mlewand** [comment]
Fixed with #2440.
